From 80bb72f2afc40a8b9f0aed102f624b693afb6178 Mon Sep 17 00:00:00 2001
From: Ashish Kumar <Ashish.Kumar@nxp.com>
Date: Wed, 3 Feb 2016 17:41:06 +0530
Subject: [PATCH 0806/1429] B4860qds/B4420qds: Add DPAA nodes in DTS files

 Add DPAA nodes in dts file for B4860qds and B4420qds

Signed-off-by: Ashish Kumar <Ashish.Kumar@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/powerpc/boot/dts/fsl/b4420qds.dts      |   15 +
 arch/powerpc/boot/dts/fsl/b4420si-post.dtsi |  192 +++++++++++++
 arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi  |    8 +
 arch/powerpc/boot/dts/fsl/b4860qds.dts      |   81 ++++++-
 arch/powerpc/boot/dts/fsl/b4860si-post.dtsi |  407 +++++++++++++++++++++++----
 arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi  |   12 +
 arch/powerpc/boot/dts/fsl/b4qds.dtsi        |   73 +++++-
 arch/powerpc/boot/dts/fsl/b4si-post.dtsi    |  152 +++++------
 8 files changed, 800 insertions(+), 140 deletions(-)

diff --git a/arch/powerpc/boot/dts/fsl/b4420qds.dts b/arch/powerpc/boot/dts/fsl/b4420qds.dts
index cd9203c..01598e7 100644
--- a/arch/powerpc/boot/dts/fsl/b4420qds.dts
+++ b/arch/powerpc/boot/dts/fsl/b4420qds.dts
@@ -45,6 +45,21 @@
 		};
 	};
 
+	fsl,dpaa {
+		compatible = "fsl,b4420-dpaa", "fsl,dpaa";
+		ethernet@0 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@1 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@2 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@3 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+	};
 };
 
 /include/ "b4420si-post.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi
index c0fe250..ec296eb2 100644
--- a/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi
@@ -34,6 +34,198 @@
 
 /include/ "b4si-post.dtsi"
 
+&bportals {
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	compatible = "simple-bus";
+	bman-portal@0 {
+		cell-index = <0x0>;
+		compatible = "fsl,bman-portal";
+		reg = <0x0 0x4000 0x1000000 0x1000>;
+		interrupts = <105 2 0 0>;
+	};
+	bman-portal@4000 {
+		cell-index = <0x1>;
+		compatible = "fsl,bman-portal";
+		reg = <0x4000 0x4000 0x1001000 0x1000>;
+		interrupts = <107 2 0 0>;
+	};
+	bman-portal@8000 {
+		cell-index = <2>;
+		compatible = "fsl,bman-portal";
+		reg = <0x8000 0x4000 0x1002000 0x1000>;
+		interrupts = <109 2 0 0>;
+	};
+	bman-portal@c000 {
+		cell-index = <0x3>;
+		compatible = "fsl,bman-portal";
+		reg = <0xc000 0x4000 0x1003000 0x1000>;
+		interrupts = <111 2 0 0>;
+	};
+	bman-portal@10000 {
+		cell-index = <0x4>;
+		compatible = "fsl,bman-portal";
+		reg = <0x10000 0x4000 0x1004000 0x1000>;
+		interrupts = <113 2 0 0>;
+	};
+	bman-portal@14000 {
+		cell-index = <0x5>;
+		compatible = "fsl,bman-portal";
+		reg = <0x14000 0x4000 0x1005000 0x1000>;
+		interrupts = <115 2 0 0>;
+	};
+	bman-portal@18000 {
+		cell-index = <0x6>;
+		compatible = "fsl,bman-portal";
+		reg = <0x18000 0x4000 0x1006000 0x1000>;
+		interrupts = <117 2 0 0>;
+	};
+	bman-portal@1c000 {
+		cell-index = <0x7>;
+		compatible = "fsl,bman-portal";
+		reg = <0x1c000 0x4000 0x1007000 0x1000>;
+		interrupts = <119 2 0 0>;
+	};
+	bman-portal@20000 {
+		cell-index = <0x8>;
+		compatible = "fsl,bman-portal";
+		reg = <0x20000 0x4000 0x1008000 0x1000>;
+		interrupts = <121 2 0 0>;
+	};
+	bman-portal@24000 {
+		cell-index = <0x9>;
+		compatible = "fsl,bman-portal";
+		reg = <0x24000 0x4000 0x1009000 0x1000>;
+		interrupts = <123 2 0 0>;
+	};
+	bman-portal@28000 {
+		cell-index = <0xa>;
+		compatible = "fsl,bman-portal";
+		reg = <0x28000 0x4000 0x100a000 0x1000>;
+		interrupts = <125 2 0 0>;
+	};
+	bman-portal@2c000 {
+		cell-index = <0xb>;
+		compatible = "fsl,bman-portal";
+		reg = <0x2c000 0x4000 0x100b000 0x1000>;
+		interrupts = <127 2 0 0>;
+	};
+	bman-portal@30000 {
+		cell-index = <0xc>;
+		compatible = "fsl,bman-portal";
+		reg = <0x30000 0x4000 0x100c000 0x1000>;
+		interrupts = <129 2 0 0>;
+	};
+	bman-portal@34000 {
+		cell-index = <0xd>;
+		compatible = "fsl,bman-portal";
+		reg = <0x34000 0x4000 0x100d000 0x1000>;
+		interrupts = <131 2 0 0>;
+	};
+};
+&qportals {
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	compatible = "simple-bus";
+	qportal0: qman-portal@0 {
+		cell-index = <0x0>;
+		compatible = "fsl,qman-portal";
+		reg = <0x0 0x4000 0x1000000 0x1000>;
+		interrupts = <104 0x2 0 0>;
+		fsl,qman-channel-id = <0x0>;
+	};
+	qportal1: qman-portal@4000 {
+		cell-index = <0x1>;
+		compatible = "fsl,qman-portal";
+		reg = <0x4000 0x4000 0x1001000 0x1000>;
+		interrupts = <106 0x2 0 0>;
+		fsl,qman-channel-id = <0x1>;
+	};
+	qportal2: qman-portal@8000 {
+		cell-index = <0x2>;
+		compatible = "fsl,qman-portal";
+		reg = <0x8000 0x4000 0x1002000 0x1000>;
+		interrupts = <108 0x2 0 0>;
+		fsl,qman-channel-id = <0x2>;
+	};
+	qportal3: qman-portal@c000 {
+		cell-index = <0x3>;
+		compatible = "fsl,qman-portal";
+		reg = <0xc000 0x4000 0x1003000 0x1000>;
+		interrupts = <110 0x2 0 0>;
+		fsl,qman-channel-id = <0x3>;
+	};
+	qportal4: qman-portal@10000 {
+		cell-index = <0x4>;
+		compatible = "fsl,qman-portal";
+		reg = <0x10000 0x4000 0x1004000 0x1000>;
+		interrupts = <112 0x2 0 0>;
+		fsl,qman-channel-id = <0x4>;
+	};
+	qportal5: qman-portal@14000 {
+		cell-index = <0x5>;
+		compatible = "fsl,qman-portal";
+		reg = <0x14000 0x4000 0x1005000 0x1000>;
+		interrupts = <114 0x2 0 0>;
+		fsl,qman-channel-id = <0x5>;
+	};
+	qportal6: qman-portal@18000 {
+		cell-index = <0x6>;
+		compatible = "fsl,qman-portal";
+		reg = <0x18000 0x4000 0x1006000 0x1000>;
+		interrupts = <116 0x2 0 0>;
+		fsl,qman-channel-id = <0x6>;
+	};
+	qportal7: qman-portal@1c000 {
+		cell-index = <0x7>;
+		compatible = "fsl,qman-portal";
+		reg = <0x1c000 0x4000 0x1007000 0x1000>;
+		interrupts = <118 0x2 0 0>;
+		fsl,qman-channel-id = <0x7>;
+	};
+	qportal8: qman-portal@20000 {
+		cell-index = <0x8>;
+		compatible = "fsl,qman-portal";
+		reg = <0x20000 0x4000 0x1008000 0x1000>;
+		interrupts = <120 0x2 0 0>;
+		fsl,qman-channel-id = <0x8>;
+	};
+	qportal9: qman-portal@24000 {
+		cell-index = <0x9>;
+		compatible = "fsl,qman-portal";
+		reg = <0x24000 0x4000 0x1009000 0x1000>;
+		interrupts = <122 0x2 0 0>;
+		fsl,qman-channel-id = <0x9>;
+	};
+	qportal10: qman-portal@28000 {
+		cell-index = <0xa>;
+		compatible = "fsl,qman-portal";
+		reg = <0x28000 0x4000 0x100a000 0x1000>;
+		interrupts = <124 0x2 0 0>;
+		fsl,qman-channel-id = <0xa>;
+	};
+	qportal11: qman-portal@2c000 {
+		cell-index = <0xb>;
+		compatible = "fsl,qman-portal";
+		reg = <0x2c000 0x4000 0x100b000 0x1000>;
+		interrupts = <126 0x2 0 0>;
+		fsl,qman-channel-id = <0xb>;
+	};
+	qportal12: qman-portal@30000 {
+		cell-index = <0xc>;
+		compatible = "fsl,qman-portal";
+		reg = <0x30000 0x4000 0x100c000 0x1000>;
+		interrupts = <128 0x2 0 0>;
+		fsl,qman-channel-id = <0xc>;
+	};
+	qportal13: qman-portal@34000 {
+		cell-index = <0xd>;
+		compatible = "fsl,qman-portal";
+		reg = <0x34000 0x4000 0x100d000 0x1000>;
+		interrupts = <130 0x2 0 0>;
+		fsl,qman-channel-id = <0xd>;
+	};
+};
 /* controller at 0x200000 */
 &pci0 {
 	compatible = "fsl,b4420-pcie", "fsl,qoriq-pcie-v2.4";
diff --git a/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi b/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi
index 29e6fc0..bf276ac 100644
--- a/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi
@@ -50,6 +50,14 @@
 		serial1 = &serial1;
 		serial2 = &serial2;
 		serial3 = &serial3;
+		crypto = &crypto;
+		qman = &qman;
+		bman = &bman;
+		fman0 = &fman0;
+		ethernet0 = &fm1mac1;
+		ethernet1 = &fm1mac2;
+		ethernet2 = &fm1mac3;
+		ethernet3 = &fm1mac4;
 		pci0 = &pci0;
 		usb0 = &usb0;
 		dma0 = &dma0;
diff --git a/arch/powerpc/boot/dts/fsl/b4860qds.dts b/arch/powerpc/boot/dts/fsl/b4860qds.dts
index ba8c9be..fb5e7b8 100644
--- a/arch/powerpc/boot/dts/fsl/b4860qds.dts
+++ b/arch/powerpc/boot/dts/fsl/b4860qds.dts
@@ -39,11 +39,60 @@
 	model = "fsl,B4860QDS";
 	compatible = "fsl,B4860QDS";
 
+	aliases{
+		phy_sgmii_1e = &phy_sgmii_1e;
+		phy_sgmii_1f = &phy_sgmii_1f;
+		phy_xaui_slot1 = &phy_xaui_slot1;
+		phy_xaui_slot2 = &phy_xaui_slot2;
+	};
 	ifc: localbus@ffe124000 {
 		board-control@3,0 {
 			compatible = "fsl,b4860qds-fpga", "fsl,fpga-qixis";
 		};
 	};
+	soc: soc@ffe000000 {
+		fman0: fman@400000 {
+			fm1mac5: ethernet@e8000 {
+				phy-handle = <&phy_sgmii_1e>;
+				phy-connection-type = "sgmii";
+			};
+			fm1mac6: ethernet@ea000 {
+				phy-handle = <&phy_sgmii_1f>;
+				phy-connection-type = "sgmii";
+			};
+			fm1mac9: ethernet@f0000 { /* FM1@TGEC2 */
+				phy-handle = <&phy_xaui_slot1>;
+				phy-connection-type = "xgmii";
+			};
+			fm1mac10: ethernet@f2000 { /* FM1@TGEC1 */
+				phy-handle = <&phy_xaui_slot2>;
+				phy-connection-type = "xgmii";
+			};
+			mdio0: mdio@fc000 {
+				phy_sgmii_1e: ethernet-phy@1e {
+					status = "disabled";
+					reg = <0x1e>;
+				};
+				phy_sgmii_1f: ethernet-phy@1f {
+					status = "disabled";
+					reg = <0x1f>;
+				};
+			};
+			xmdio0: mdio@fd000 {
+				/* For 10g interfaces */
+				phy_xaui_slot1: xaui-phy@slot1 {
+					status = "disabled";
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x7>; /* default switch setting on slot1 of AMC2PEX */
+				};
+				phy_xaui_slot2: xaui-phy@slot2 {
+					status = "disabled";
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x6>; /* default switch setting on slot1 of AMC2PEX */
+				};
+			};
+		};
+	};
 
 	rio: rapidio@ffe0c0000 {
 		reg = <0xf 0xfe0c0000 0 0x11000>;
@@ -55,7 +104,37 @@
 			ranges = <0 0 0xc 0x30000000 0 0x10000000>;
 		};
 	};
-
+	fsl,dpaa {
+		compatible = "fsl,b4860-dpaa", "fsl,dpaa";
+		ethernet@0 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@1 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@2 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@3 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@4 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac5>;
+		};
+		ethernet@5 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac6>;
+		};
+		ethernet@8 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac9>;
+		};
+		ethernet@9 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac10>;
+		};
+	};
 };
 
 /include/ "b4860si-post.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
index 536d152..ae7ca57 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
@@ -33,7 +33,342 @@
  */
 
 /include/ "b4si-post.dtsi"
+&bportals {
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	compatible = "simple-bus";
+	bman-portal@0 {
+		cell-index = <0x0>;
+		compatible = "fsl,bman-portal";
+		reg = <0x0 0x4000 0x1000000 0x1000>;
+		interrupts = <105 2 0 0>;
+	};
+	bman-portal@4000 {
+		cell-index = <0x1>;
+		compatible = "fsl,bman-portal";
+		reg = <0x4000 0x4000 0x1001000 0x1000>;
+		interrupts = <107 2 0 0>;
+	};
+	bman-portal@8000 {
+		cell-index = <2>;
+		compatible = "fsl,bman-portal";
+		reg = <0x8000 0x4000 0x1002000 0x1000>;
+		interrupts = <109 2 0 0>;
+	};
+	bman-portal@c000 {
+		cell-index = <0x3>;
+		compatible = "fsl,bman-portal";
+		reg = <0xc000 0x4000 0x1003000 0x1000>;
+		interrupts = <111 2 0 0>;
+	};
+	bman-portal@10000 {
+		cell-index = <0x4>;
+		compatible = "fsl,bman-portal";
+		reg = <0x10000 0x4000 0x1004000 0x1000>;
+		interrupts = <113 2 0 0>;
+	};
+	bman-portal@14000 {
+		cell-index = <0x5>;
+		compatible = "fsl,bman-portal";
+		reg = <0x14000 0x4000 0x1005000 0x1000>;
+		interrupts = <115 2 0 0>;
+	};
+	bman-portal@18000 {
+		cell-index = <0x6>;
+		compatible = "fsl,bman-portal";
+		reg = <0x18000 0x4000 0x1006000 0x1000>;
+		interrupts = <117 2 0 0>;
+	};
+	bman-portal@1c000 {
+		cell-index = <0x7>;
+		compatible = "fsl,bman-portal";
+		reg = <0x1c000 0x4000 0x1007000 0x1000>;
+		interrupts = <119 2 0 0>;
+	};
+	bman-portal@20000 {
+		cell-index = <0x8>;
+		compatible = "fsl,bman-portal";
+		reg = <0x20000 0x4000 0x1008000 0x1000>;
+		interrupts = <121 2 0 0>;
+	};
+	bman-portal@24000 {
+		cell-index = <0x9>;
+		compatible = "fsl,bman-portal";
+		reg = <0x24000 0x4000 0x1009000 0x1000>;
+		interrupts = <123 2 0 0>;
+	};
+	bman-portal@28000 {
+		cell-index = <0xa>;
+		compatible = "fsl,bman-portal";
+		reg = <0x28000 0x4000 0x100a000 0x1000>;
+		interrupts = <125 2 0 0>;
+	};
+	bman-portal@2c000 {
+		cell-index = <0xb>;
+		compatible = "fsl,bman-portal";
+		reg = <0x2c000 0x4000 0x100b000 0x1000>;
+		interrupts = <127 2 0 0>;
+	};
+	bman-portal@30000 {
+		cell-index = <0xc>;
+		compatible = "fsl,bman-portal";
+		reg = <0x30000 0x4000 0x100c000 0x1000>;
+		interrupts = <129 2 0 0>;
+	};
+	bman-portal@34000 {
+		cell-index = <0xd>;
+		compatible = "fsl,bman-portal";
+		reg = <0x34000 0x4000 0x100d000 0x1000>;
+		interrupts = <131 2 0 0>;
+	};
+	bman-portal@38000 {
+		cell-index = <0xe>;
+		compatible = "fsl,bman-portal";
+		reg = <0x38000 0x4000 0x100e000 0x1000>;
+		interrupts = <133 2 0 0>;
+	};
+	bman-portal@3c000 {
+		cell-index = <0xf>;
+		compatible = "fsl,bman-portal";
+		reg = <0x3c000 0x4000 0x100f000 0x1000>;
+		interrupts = <135 2 0 0>;
+	};
+	bman-portal@40000 {
+		cell-index = <0x10>;
+		compatible = "fsl,bman-portal";
+		reg = <0x40000 0x4000 0x1010000 0x1000>;
+		interrupts = <137 2 0 0>;
+	};
+	bman-portal@44000 {
+		cell-index = <0x11>;
+		compatible = "fsl,bman-portal";
+		reg = <0x44000 0x4000 0x1011000 0x1000>;
+		interrupts = <139 2 0 0>;
+	};
+	bman-portal@48000 {
+		cell-index = <0x12>;
+		compatible = "fsl,bman-portal";
+		reg = <0x48000 0x4000 0x1012000 0x1000>;
+		interrupts = <141 2 0 0>;
+	};
+	bman-portal@4c000 {
+		cell-index = <0x13>;
+		compatible = "fsl,bman-portal";
+		reg = <0x4c000 0x4000 0x1013000 0x1000>;
+		interrupts = <143 2 0 0>;
+	};
+	bman-portal@50000 {
+		cell-index = <0x14>;
+		compatible = "fsl,bman-portal";
+		reg = <0x50000 0x4000 0x1014000 0x1000>;
+		interrupts = <145 2 0 0>;
+	};
+	bman-portal@54000 {
+		cell-index = <0x15>;
+		compatible = "fsl,bman-portal";
+		reg = <0x54000 0x4000 0x1015000 0x1000>;
+		interrupts = <147 2 0 0>;
+	};
+	bman-portal@58000 {
+		cell-index = <0x16>;
+		compatible = "fsl,bman-portal";
+		reg = <0x58000 0x4000 0x1016000 0x1000>;
+		interrupts = <149 2 0 0>;
+	};
+	bman-portal@5c000 {
+		cell-index = <0x17>;
+		compatible = "fsl,bman-portal";
+		reg = <0x5c000 0x4000 0x1017000 0x1000>;
+		interrupts = <151 2 0 0>;
+	};
+	bman-portal@60000 {
+		cell-index = <0x18>;
+		compatible = "fsl,bman-portal";
+		reg = <0x60000 0x4000 0x1018000 0x1000>;
+		interrupts = <153 2 0 0>;
+	};
+};
 
+&qportals {
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	compatible = "simple-bus";
+	qportal0: qman-portal@0 {
+		cell-index = <0x0>;
+		compatible = "fsl,qman-portal";
+		reg = <0x0 0x4000 0x1000000 0x1000>;
+		interrupts = <104 0x2 0 0>;
+		fsl,qman-channel-id = <0x0>;
+	};
+	qportal1: qman-portal@4000 {
+		cell-index = <0x1>;
+		compatible = "fsl,qman-portal";
+		reg = <0x4000 0x4000 0x1001000 0x1000>;
+		interrupts = <106 0x2 0 0>;
+		fsl,qman-channel-id = <0x1>;
+	};
+	qportal2: qman-portal@8000 {
+		cell-index = <0x2>;
+		compatible = "fsl,qman-portal";
+		reg = <0x8000 0x4000 0x1002000 0x1000>;
+		interrupts = <108 0x2 0 0>;
+		fsl,qman-channel-id = <0x2>;
+	};
+	qportal3: qman-portal@c000 {
+		cell-index = <0x3>;
+		compatible = "fsl,qman-portal";
+		reg = <0xc000 0x4000 0x1003000 0x1000>;
+		interrupts = <110 0x2 0 0>;
+		fsl,qman-channel-id = <0x3>;
+	};
+	qportal4: qman-portal@10000 {
+		cell-index = <0x4>;
+		compatible = "fsl,qman-portal";
+		reg = <0x10000 0x4000 0x1004000 0x1000>;
+		interrupts = <112 0x2 0 0>;
+		fsl,qman-channel-id = <0x4>;
+	};
+	qportal5: qman-portal@14000 {
+		cell-index = <0x5>;
+		compatible = "fsl,qman-portal";
+		reg = <0x14000 0x4000 0x1005000 0x1000>;
+		interrupts = <114 0x2 0 0>;
+		fsl,qman-channel-id = <0x5>;
+	};
+	qportal6: qman-portal@18000 {
+		cell-index = <0x6>;
+		compatible = "fsl,qman-portal";
+		reg = <0x18000 0x4000 0x1006000 0x1000>;
+		interrupts = <116 0x2 0 0>;
+		fsl,qman-channel-id = <0x6>;
+	};
+	qportal7: qman-portal@1c000 {
+		cell-index = <0x7>;
+		compatible = "fsl,qman-portal";
+		reg = <0x1c000 0x4000 0x1007000 0x1000>;
+		interrupts = <118 0x2 0 0>;
+		fsl,qman-channel-id = <0x7>;
+	};
+	qportal8: qman-portal@20000 {
+		cell-index = <0x8>;
+		compatible = "fsl,qman-portal";
+		reg = <0x20000 0x4000 0x1008000 0x1000>;
+		interrupts = <120 0x2 0 0>;
+		fsl,qman-channel-id = <0x8>;
+	};
+	qportal9: qman-portal@24000 {
+		cell-index = <0x9>;
+		compatible = "fsl,qman-portal";
+		reg = <0x24000 0x4000 0x1009000 0x1000>;
+		interrupts = <122 0x2 0 0>;
+		fsl,qman-channel-id = <0x9>;
+	};
+	qportal10: qman-portal@28000 {
+		cell-index = <0xa>;
+		compatible = "fsl,qman-portal";
+		reg = <0x28000 0x4000 0x100a000 0x1000>;
+		interrupts = <124 0x2 0 0>;
+		fsl,qman-channel-id = <0xa>;
+	};
+	qportal11: qman-portal@2c000 {
+		cell-index = <0xb>;
+		compatible = "fsl,qman-portal";
+		reg = <0x2c000 0x4000 0x100b000 0x1000>;
+		interrupts = <126 0x2 0 0>;
+		fsl,qman-channel-id = <0xb>;
+	};
+	qportal12: qman-portal@30000 {
+		cell-index = <0xc>;
+		compatible = "fsl,qman-portal";
+		reg = <0x30000 0x4000 0x100c000 0x1000>;
+		interrupts = <128 0x2 0 0>;
+		fsl,qman-channel-id = <0xc>;
+	};
+	qportal13: qman-portal@34000 {
+		cell-index = <0xd>;
+		compatible = "fsl,qman-portal";
+		reg = <0x34000 0x4000 0x100d000 0x1000>;
+		interrupts = <130 0x2 0 0>;
+		fsl,qman-channel-id = <0xd>;
+	};
+	qportal14: qman-portal@38000 {
+		cell-index = <0xe>;
+		compatible = "fsl,qman-portal";
+		reg = <0x38000 0x4000 0x100e000 0x1000>;
+		interrupts = <132 0x2 0 0>;
+		fsl,qman-channel-id = <0xe>;
+	};
+	qportal15: qman-portal@3c000 {
+		cell-index = <0xf>;
+		compatible = "fsl,qman-portal";
+		reg = <0x3c000 0x4000 0x100f000 0x1000>;
+		interrupts = <134 0x2 0 0>;
+		fsl,qman-channel-id = <0xf>;
+	};
+	qportal16: qman-portal@40000 {
+		cell-index = <0x10>;
+		compatible = "fsl,qman-portal";
+		reg = <0x40000 0x4000 0x1010000 0x1000>;
+		interrupts = <136 0x2 0 0>;
+		fsl,qman-channel-id = <0x10>;
+	};
+	qportal17: qman-portal@44000 {
+		cell-index = <0x11>;
+		compatible = "fsl,qman-portal";
+		reg = <0x44000 0x4000 0x1011000 0x1000>;
+		interrupts = <138 0x2 0 0>;
+		fsl,qman-channel-id = <0x11>;
+	};
+	qportal18: qman-portal@48000 {
+		cell-index = <0x12>;
+		compatible = "fsl,qman-portal";
+		reg = <0x48000 0x4000 0x1012000 0x1000>;
+		interrupts = <140 0x2 0 0>;
+		fsl,qman-channel-id = <0x12>;
+	};
+	qportal19: qman-portal@4c000 {
+		cell-index = <0x13>;
+		compatible = "fsl,qman-portal";
+		reg = <0x4c000 0x4000 0x1013000 0x1000>;
+		interrupts = <142 0x2 0 0>;
+		fsl,qman-channel-id = <0x13>;
+	};
+	qportal20: qman-portal@50000 {
+		cell-index = <0x14>;
+		compatible = "fsl,qman-portal";
+		reg = <0x50000 0x4000 0x1014000 0x1000>;
+		interrupts = <144 0x2 0 0>;
+		fsl,qman-channel-id = <0x14>;
+	};
+	qportal21: qman-portal@54000 {
+		cell-index = <0x15>;
+		compatible = "fsl,qman-portal";
+		reg = <0x54000 0x4000 0x1015000 0x1000>;
+		interrupts = <146 0x2 0 0>;
+		fsl,qman-channel-id = <0x15>;
+	};
+	qportal22: qman-portal@58000 {
+		cell-index = <0x16>;
+		compatible = "fsl,qman-portal";
+		reg = <0x58000 0x4000 0x1016000 0x1000>;
+		interrupts = <148 0x2 0 0>;
+		fsl,qman-channel-id = <0x16>;
+	};
+	qportal23: qman-portal@5c000 {
+		cell-index = <0x17>;
+		compatible = "fsl,qman-portal";
+		reg = <0x5c000 0x4000 0x1017000 0x1000>;
+		interrupts = <150 0x2 0 0>;
+		fsl,qman-channel-id = <0x17>;
+	};
+	qportal24: qman-portal@60000 {
+		cell-index = <0x18>;
+		compatible = "fsl,qman-portal";
+		reg = <0x60000 0x4000 0x1018000 0x1000>;
+		interrupts = <152 0x2 0 0>;
+		fsl,qman-channel-id = <0x18>;
+	};
+};
 /* controller at 0x200000 */
 &pci0 {
 	compatible = "fsl,b4860-pcie", "fsl,qoriq-pcie-v2.4";
@@ -109,63 +444,6 @@
 	};
 };
 
-&bportals {
-	bman-portal@38000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x38000 0x4000>, <0x100e000 0x1000>;
-		interrupts = <133 2 0 0>;
-	};
-	bman-portal@3c000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x3c000 0x4000>, <0x100f000 0x1000>;
-		interrupts = <135 2 0 0>;
-	};
-	bman-portal@40000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x40000 0x4000>, <0x1010000 0x1000>;
-		interrupts = <137 2 0 0>;
-	};
-	bman-portal@44000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x44000 0x4000>, <0x1011000 0x1000>;
-		interrupts = <139 2 0 0>;
-	};
-	bman-portal@48000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x48000 0x4000>, <0x1012000 0x1000>;
-		interrupts = <141 2 0 0>;
-	};
-	bman-portal@4c000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x4c000 0x4000>, <0x1013000 0x1000>;
-		interrupts = <143 2 0 0>;
-	};
-	bman-portal@50000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x50000 0x4000>, <0x1014000 0x1000>;
-		interrupts = <145 2 0 0>;
-	};
-	bman-portal@54000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x54000 0x4000>, <0x1015000 0x1000>;
-		interrupts = <147 2 0 0>;
-	};
-	bman-portal@58000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x58000 0x4000>, <0x1016000 0x1000>;
-		interrupts = <149 2 0 0>;
-	};
-	bman-portal@5c000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x5c000 0x4000>, <0x1017000 0x1000>;
-		interrupts = <151 2 0 0>;
-	};
-	bman-portal@60000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x60000 0x4000>, <0x1018000 0x1000>;
-		interrupts = <153 2 0 0>;
-	};
-};
 
 &soc {
 	ddr2: memory-controller@9000 {
@@ -178,6 +456,21 @@
 		compatible = "fsl,b4860-l3-cache-controller", "cache";
 	};
 
+
+/include/ "qoriq-fman3-0-1g-4.dtsi"
+/include/ "qoriq-fman3-0-1g-5.dtsi"
+/include/ "qoriq-fman3-0-10g-0.dtsi"
+/include/ "qoriq-fman3-0-10g-1.dtsi"
+	fman0: fman@400000 {
+		/* tx - 1g - 4 */
+		port@ac000 {
+			fsl,qman-channel-id = <0x806>;
+		};
+		/* tx - 1g - 5 */
+		port@ad000 {
+			fsl,qman-channel-id = <0x807>;
+		};
+	};
 	guts: global-utilities@e0000 {
 		compatible = "fsl,b4860-device-config", "fsl,qoriq-device-config-2.0";
 	};
diff --git a/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
index f48662e..32edec5 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
@@ -50,6 +50,18 @@
 		serial1 = &serial1;
 		serial2 = &serial2;
 		serial3 = &serial3;
+		crypto = &crypto;
+		qman = &qman;
+		bman = &bman;
+		fman0 = &fman0;
+		ethernet0 = &fm1mac1;
+		ethernet1 = &fm1mac2;
+		ethernet2 = &fm1mac3;
+		ethernet3 = &fm1mac4;
+		ethernet4 = &fm1mac5;
+		ethernet5 = &fm1mac6;
+		ethernet8 = &fm1mac9;
+		ethernet9 = &fm1mac10;
 		pci0 = &pci0;
 		usb0 = &usb0;
 		dma0 = &dma0;
diff --git a/arch/powerpc/boot/dts/fsl/b4qds.dtsi b/arch/powerpc/boot/dts/fsl/b4qds.dtsi
index 6e23470..c144449 100644
--- a/arch/powerpc/boot/dts/fsl/b4qds.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4qds.dtsi
@@ -38,6 +38,12 @@
 	#address-cells = <2>;
 	#size-cells = <2>;
 	interrupt-parent = <&mpic>;
+	aliases{
+		phy_sgmii_10 = &phy_sgmii_10;
+		phy_sgmii_11 = &phy_sgmii_11;
+		phy_sgmii_1c = &phy_sgmii_1c;
+		phy_sgmii_1d = &phy_sgmii_1d;
+	};
 
 	ifc: localbus@ffe124000 {
 		reg = <0xf 0xfe124000 0 0x2000>;
@@ -101,13 +107,19 @@
 		#address-cells = <2>;
 		#size-cells = <2>;
 		ranges;
-
 		bman_fbpr: bman-fbpr {
 			size = <0 0x1000000>;
 			alignment = <0 0x1000000>;
 		};
+		qman_fqd: qman-fqd {
+			size = <0 0x400000>;
+			alignment = <0 0x400000>;
+		};
+		qman_pfdr: qman-pfdr {
+			size = <0 0x2000000>;
+			alignment = <0 0x2000000>;
+		};
 	};
-
 	dcsr: dcsr@f00000000 {
 		ranges = <0x00000000 0xf 0x00000000 0x01052000>;
 	};
@@ -115,6 +127,9 @@
 	bportals: bman-portals@ff4000000 {
 		ranges = <0x0 0xf 0xf4000000 0x2000000>;
 	};
+	qportals: qman-portals@ff6000000 {
+		ranges = <0x0 0xf 0xf6000000 0x2000000>;
+	};
 
 	soc: soc@ffe000000 {
 		ranges = <0x00000000 0xf 0xfe000000 0x1000000>;
@@ -198,6 +213,40 @@
 			phy_type = "ulpi";
 		};
 
+		fman0: fman@400000 {
+			fm1mac1: ethernet@e0000 {
+				phy-handle = <&phy_sgmii_10>;
+				phy-connection-type = "sgmii";
+			};
+			fm1mac2: ethernet@e2000 {
+				phy-handle = <&phy_sgmii_11>;
+				phy-connection-type = "sgmii";
+			};
+			fm1mac3: ethernet@e4000 {
+				phy-handle = <&phy_sgmii_1c>;
+				phy-connection-type = "sgmii";
+			};
+			fm1mac4: ethernet@e6000 {
+				phy-handle = <&phy_sgmii_1d>;
+				phy-connection-type = "sgmii";
+			};
+			mdio0: mdio@fc000 {
+				phy_sgmii_10: ethernet-phy@10 {
+					reg = <0x10>;
+				};
+				phy_sgmii_11: ethernet-phy@11 {
+					reg = <0x11>;
+				};
+				phy_sgmii_1c: ethernet-phy@1c {
+					status = "disabled";
+					reg = <0x1c>;
+				};
+				phy_sgmii_1d: ethernet-phy@1d {
+					status = "disabled";
+					reg = <0x1d>;
+				};
+			};
+		};
 	};
 
 	pci0: pcie@ffe200000 {
@@ -215,7 +264,27 @@
 		};
 	};
 
+	fsl,dpaa {
+		compatible = "fsl,dpaa";
+		ethernet@0 {
+			compatible = "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac1>;
+		};
+		ethernet@1 {
+			compatible = "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac2>;
+		};
+		ethernet@2 {
+			compatible = "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac3>;
+		};
+		ethernet@3 {
+			compatible = "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac4>;
+		};
+	};
 };
 
 /include/ "b4si-post.dtsi"
+/include/ "qoriq-dpaa-res3.dtsi"
 /include/ "qoriq-qman-ceetm0.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/b4si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
index 350ea50..2240840 100644
--- a/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
@@ -36,6 +36,14 @@
 	compatible = "fsl,bman-fbpr";
 	alloc-ranges = <0 0 0x10000 0>;
 };
+&qman_fqd {
+	compatible = "fsl,qman-fqd";
+	alloc-ranges = <0 0 0x10000 0>;
+};
+&qman_pfdr {
+	compatible = "fsl,qman-pfdr";
+	alloc-ranges = <0 0 0x10000 0>;
+};
 
 &ifc {
 	#address-cells = <2>;
@@ -133,82 +141,6 @@
 	};
 };
 
-&bportals {
-	#address-cells = <0x1>;
-	#size-cells = <0x1>;
-	compatible = "simple-bus";
-
-	bman-portal@0 {
-		compatible = "fsl,bman-portal";
-		reg = <0x0 0x4000>, <0x1000000 0x1000>;
-		interrupts = <105 2 0 0>;
-	};
-	bman-portal@4000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x4000 0x4000>, <0x1001000 0x1000>;
-		interrupts = <107 2 0 0>;
-	};
-	bman-portal@8000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x8000 0x4000>, <0x1002000 0x1000>;
-		interrupts = <109 2 0 0>;
-	};
-	bman-portal@c000 {
-		compatible = "fsl,bman-portal";
-		reg = <0xc000 0x4000>, <0x1003000 0x1000>;
-		interrupts = <111 2 0 0>;
-	};
-	bman-portal@10000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x10000 0x4000>, <0x1004000 0x1000>;
-		interrupts = <113 2 0 0>;
-	};
-	bman-portal@14000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x14000 0x4000>, <0x1005000 0x1000>;
-		interrupts = <115 2 0 0>;
-	};
-	bman-portal@18000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x18000 0x4000>, <0x1006000 0x1000>;
-		interrupts = <117 2 0 0>;
-	};
-	bman-portal@1c000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x1c000 0x4000>, <0x1007000 0x1000>;
-		interrupts = <119 2 0 0>;
-	};
-	bman-portal@20000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x20000 0x4000>, <0x1008000 0x1000>;
-		interrupts = <121 2 0 0>;
-	};
-	bman-portal@24000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x24000 0x4000>, <0x1009000 0x1000>;
-		interrupts = <123 2 0 0>;
-	};
-	bman-portal@28000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x28000 0x4000>, <0x100a000 0x1000>;
-		interrupts = <125 2 0 0>;
-	};
-	bman-portal@2c000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x2c000 0x4000>, <0x100b000 0x1000>;
-		interrupts = <127 2 0 0>;
-	};
-	bman-portal@30000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x30000 0x4000>, <0x100c000 0x1000>;
-		interrupts = <129 2 0 0>;
-	};
-	bman-portal@34000 {
-		compatible = "fsl,bman-portal";
-		reg = <0x34000 0x4000>, <0x100d000 0x1000>;
-		interrupts = <131 2 0 0>;
-	};
-};
 
 &soc {
 	#address-cells = <1>;
@@ -287,7 +219,70 @@
 		};
 	};
 
+/include/ "qoriq-qman3.dtsi"
+	qman: qman@318000 {
+		interrupts = <16 2 1 28>;
+	};
+/include/ "qoriq-bman1.dtsi"
+	bman: bman@31a000 {
+		interrupts = <16 2 1 29>;
+	};
+/include/ "qoriq-rman-0.dtsi"
+	rman: rman@1e0000 {
+		fsl,qman-channels-id = <0x820 0x821>;
+	};
 /include/ "qoriq-mpic4.3.dtsi"
+/include/ "qoriq-fman3-0.dtsi"
+/include/ "qoriq-fman3-0-1g-0.dtsi"
+/include/ "qoriq-fman3-0-1g-1.dtsi"
+/include/ "qoriq-fman3-0-1g-2.dtsi"
+/include/ "qoriq-fman3-0-1g-3.dtsi"
+	fman0: fman@400000 {
+		interrupts = <
+			96 2 0 0
+			16 2 1 30>;
+		/* tx - 1g - 0 */
+		port@a8000 {
+			fsl,qman-channel-id = <0x802>;
+		};
+		/* tx - 1g - 1 */
+		port@a9000 {
+			fsl,qman-channel-id = <0x803>;
+		};
+		/* tx - 1g - 2 */
+		port@aa000 {
+			fsl,qman-channel-id = <0x804>;
+		};
+		/* tx - 1g - 3 */
+		port@ab000 {
+			fsl,qman-channel-id = <0x805>;
+		};
+		/* offline - 0 is not usable on B4860 */
+		/* offline - 1 */
+		port@82000 {
+			fsl,qman-channel-id = <0x809>;
+		};
+		/* offline - 2 */
+		port@83000 {
+			fsl,qman-channel-id = <0x80a>;
+		};
+		/* offline - 3 */
+		port@84000 {
+			fsl,qman-channel-id = <0x80b>;
+		};
+		/* offline - 4 */
+		port@85000 {
+			fsl,qman-channel-id = <0x80c>;
+		};
+		/* offline - 5 */
+		port@86000 {
+			fsl,qman-channel-id = <0x80d>;
+		};
+		/* offline - 6 */
+		port@87000 {
+			fsl,qman-channel-id = <0x80e>;
+		};
+	};
 
 	guts: global-utilities@e0000 {
 		compatible = "fsl,b4-device-config";
@@ -343,9 +338,6 @@
 /include/ "qoriq-duart-1.dtsi"
 /include/ "qoriq-sec5.3-0.dtsi"
 
-/include/ "qoriq-bman1.dtsi"
-	bman: bman@31a000 {
-		interrupts = <16 2 1 29>;
-	};
+
 
 };
-- 
1.7.5.4

